packages/joy-utils/src/Transport.ts

```diff
@@ -1,6 +1,9 @@
 export abstract class Transport {
-  protected promise<T>(value: T): Promise<T> {
-    return new Promise<T>((resolve, reject) => {
+  protected async promise<T>(value: T, timeout:number = 1): Promise<T> {
```

Comment:
What about this? Note `?` on `timeout` argument and added `timeout &&` in the `if` clause.
```suggestion
  protected async promise<T>(value: T, timeout?: number): Promise<T> {
    // ...
    if (timeout && timeout > 0) {
```